### DISCUSSION TITLE
GM.xmlHttpRequest with blob: URLs from GM.getResourceUrl always return undefined

### DISCUSSION THREAD

**Discussion by Sv443** [violentmonkey/violentmonkey#2123] [Q&A] [answered]
I recently tried to use GM.xmlHttpRequest to fetch a resource via the blob: URL provided by GM.getResourceUrl but the onload callback always returns undefined, so I had to switch to the native fetch API.  
My theory is that blob: URLs belong to the website itself and cannot be accessed by the extension's GM.xmlHttpRequest API? I just wanted to make sure that that is actually the case and is not a bug.

**Comment by tophf** [comment, marked as answer]
It should be fixed in the [new version](https://github.com/violentmonkey/violentmonkey/releases).

**Reply by Sv443** [reply]
Alright, thanks a lot!

**Reply by tophf** [reply]
The built-in `fetch` is the preferred method though.
